holidays/countries/tuvalu.py

```diff
@@ -0,0 +1,201 @@
+      - `Public Holidays (Amendment) Act 1990 <https://www.paclii.org/cgi-bin/sinodisp/tv/legis/num_act/pha1990243/pha1990243.html>`_
+      - `Public Holidays Act 1 <https://tuvalu-legislation.tv/cms/images/LEGISLATION/PRINCIPAL/1937/1937-0005/1937-0005_1.pdf>`_
+      - `Labour and Employment Relations Act 2017 <https://tuvalu-legislation.tv/cms/images/LEGISLATION/PRINCIPAL/2017/2017-0014/2017-0014_1.pdf>`_
+      - `Public Holidays (Amendment) Act 2018 <https://tuvalu-legislation.tv/cms/images/LEGISLATION/AMENDING/Public%20Holidays%20(Amendment)%20Act/Public%20Holidays%20(Amendment)%20Act%202018/Public%20Holidays%20(Amendment)%20Act%202018.pdf>`_
+      - `Public Holidays (Amendment) Act 2020 <https://tuvalu-legislation.tv/cms/images/LEGISLATION/AMENDING/2020/2020-0013/2020-0013.pdf>`_
+      - `Public Holidays Act 2 <https://tuvalu-legislation.tv/cms/images/LEGISLATION/PRINCIPAL/1937/1937-0005/1937-0005_2.pdf>`_
+      - `Codes for the representation of names of countries and their subdivisions <https://www.iso.org/obp/ui/#iso:code:3166:TV>`_
+      - `TUVALU-NEWS.TV <https://web.archive.org/web/20140915180104/http://www.tuvalu-news.tv/archives/2007/01/island_special_public_holidays.html>`_
+    """
+
+    country = "TV"
+    default_language = "tvl_TV"
+    # %s (estimated).
+    estimated_label = tr("%s (fakaaoga)")
+    # %s (observed).
+    observed_label = tr("%s (fakamatakuga)")
+    # %s (observed, estimated)
+    observed_estimated_label = "%s (fakamatakuga, fakaaoga)"
+    supported_languages = ("en_GB", "en_US", "tvl_TV")
+    subdivisions = (
+        "FUN",  # Funafuti.
+        "NMG",  # Nanumaga.
+        "NMA",  # Nanumea.
+        "NIT",  # Niutao.
+        "NUI",  # Nui.
+        "NKF",  # Nukufetau.
+        "NKL",  # Nukulaelae.
+        "VAI",  # Vaitupu.
+    )
+    subdivisions_aliases = {
+        # Town/Island Councils.
+        "Funafuti": "FUN",
+        "Nanumaga": "NMG",
+        "Nanumea": "NMA",
+        "Niutao": "NIT",
+        "Nui": "NUI",
+        "Nukufetau": "NKF",
+        "Nukulaelae": "NKL",
+        "Vaitupu": "VAI",
+    }
+    # Tuvalu became fully independent of the United Kingdom on October 1, 1978
+    # Tuvalu's PUBLIC HOLIDAYS (AMENDMENT) ACT 1990 (Act 2 of 1990)
+    # It was first proclaimed on FEB 7th, 1990
+    start_year = 1990
+
+    def __init__(self, *args, **kwargs):
+        ChristianHolidays.__init__(self)
+        InternationalHolidays.__init__(self)
+        kwargs.setdefault("observed_rule", SAT_SUN_TO_NEXT_MON)
+        super().__init__(*args, **kwargs)
+
+    def _populate_public_holidays(self):
+        # Public Holidays
+
+        # New Year's Day.
+        self._add_new_years_day(tr("Tausaga Fou"))
+
+        # Commonwealth Day.
+        self._add_holiday_2nd_mon_of_mar(tr("Aso Atefenua"))
+
+        # Gospel Day.
+        self._add_holiday_2nd_mon_of_may(tr("Te Aso o te Tala Lei"))
```

Comment:
```suggestion
        self._add_holiday_1_day_past_2nd_sun_of_may(tr("Te Aso o te Tala Lei"))
```
Monday following the second Sunday in May != Second Monday in May 😉